Sources/MongoSwift/MongoClient.swift

```diff
@@ -283,6 +283,45 @@ public class MongoClient {
     /// Handlers for SDAM monitoring events.
     internal var sdamEventHandlers: [SDAMEventHandler]
 
+    private var _commandEvents: Any?
+    private var _sdamEvents: Any?
+
+#if compiler(>=5.5) && canImport(_Concurrency)
+    /// Async way to monitor command events. Need macOS above 10.15 to use practically.
+    @available(macOS 10.15, *)
+    public var commandEvents: CommandEventStream {
+        if self._commandEvents == nil {
+            self._commandEvents = CommandEventStream(stream:
+                AsyncStream { con in
+                    self.addCommandEventHandler { event in
```

Comment:
> Since the client currently owns the handler, users will just get the same one next time they invoke the computed property, even if they drop it.

ah right, yeah I think I wrote that comment before realizing in my next comment below that there was only ever one of each stream. 

yeah, I think overall as long as there aren't any technical limitations to it, the on-demand/factory approach would be a bit nicer so we can allow multiple streams, and can support stopping/restarting monitoring at will.

Reply:
Im not too sure what this means practically. From what I glean, wrapping the `add_EventHandler` means that we hold on to that callback even after the stream is gone and we want a `WeakEventHandler` instance to ensure everything is referenced correctly. 

@patrickfreed - What is a factory/on-demand approach? Kinda confused on how that would operate given the weirdness of compiler directives and the stuff that we had to navigate last week. Can discuss next week too just wanted to voice my confusion a little bit